### PR TITLE
[9.0][FIX][stock_inventory_lockdown] not allowing adjustments in locations with negative quants

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
We found that the current design was not allowing to perform inventory adjustments on a location that had negative quants.

We have done a refactoring of the module to perform lockdown validations using constraint in stock move instead of checks in quants. This will allow for a more robust lockdown mechanism, and still make it possible to perform inventory adjustments in the locked location. Also resolves an outstanding issue related to the previous design not allowing inventory adjustments where negative quants existed.

cc @StefanRijnhart @Cedric-Pigeon @lmignon @clonedagain

### CODE REVIEW

**Review by LoisRForgeFlow — APPROVED** [review]
Code + Functional review :+1:

**Review by StefanRijnhart — APPROVED** [review]
Nice. Interesting use of api.constrains!

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional test

File: stock_inventory_lockdown/models/stock_move.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from openerp import models, api, _
+from openerp.exceptions import ValidationError
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.constrains('location_dest_id', 'location_id', 'state')
+    def _check_locked_location(self):
+        for move in self:
```
Comment (rousseldenis) [inline stock_inventory_lockdown/models/stock_move.py]: @jbeficent Maybe `for move in self.filtered(lambda m: m.state != 'draft'):`

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review, minor comment

**Comment by MiquelRForgeFlow** [comment]
@rousseldenis done 😉

**Comment by LoisRForgeFlow** [comment]
@rousseldenis could you update your review?

**Review by rousseldenis — APPROVED** [review]
:+1: Code review

**Pull request merged**